src/components/personalInfo/personalInfo.tsx

```diff
@@ -0,0 +1,251 @@
+import React, { useEffect, useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { RootState, AppDispatch } from "../../redux/store";
+import { fetchUser } from "../../redux/slices/userSlices";
+import "./personalInfoStyles.scss";
+import { useParams } from "react-router-dom";
+import { format } from "date-fns";
+import Modal from "../modal/modal";
+import bgPhoto from "../../assets/images/bg.png";
+import nodata from "../../assets/images/nodata.png"
+import { Link } from "react-router-dom";
+import UpdateProfile from "../../views/updateprofile";
+
+const PersonalInfo: React.FC = () => {
+  const dispatch = useDispatch<AppDispatch>();
+  const { user, loading, error, isSuccess } = useSelector(
+    (state: RootState) => state.user
+  );
+  const { id } = useParams<{ id?: string }>();
```

Comment:
Consider checking why this is still undefined even when the login has succeeded.